app/serializers/species_serializer.rb

```diff
@@ -189,26 +194,23 @@ def edible
   # end
 
   def images
-    {
-      flower: Panko::ArraySerializer.new(object.species_images.flower_part, each_serializer: SpeciesImageSerializer).to_a,
-      leaf: Panko::ArraySerializer.new(object.species_images.leaf_part, each_serializer: SpeciesImageSerializer).to_a,
-      habit: Panko::ArraySerializer.new(object.species_images.habit_part, each_serializer: SpeciesImageSerializer).to_a,
-      fruit: Panko::ArraySerializer.new(object.species_images.fruit_part, each_serializer: SpeciesImageSerializer).to_a,
-      bark: Panko::ArraySerializer.new(object.species_images.bark_part, each_serializer: SpeciesImageSerializer).to_a,
-      other: Panko::ArraySerializer.new(object.species_images.other_part, each_serializer: SpeciesImageSerializer).to_a
-    }
+    object.species_images.group_by(&:part).transform_values do |images|
+      Panko::ArraySerializer.new(images, each_serializer: SpeciesImageSerializer).to_a
+    end
   end
 
   def common_names
     object.common_names.group_by(&:lang).transform_values {|e| e.pluck(:name) }
   end
 
   def distribution
-    object.species_distributions.joins(:zone).group_by(&:establishment).transform_values {|e| e.map {|r| r.zone.name } }
+    object.species_distributions.includes(:zone).group_by(&:establishment).transform_values do |e|
```

Comment:
This appears to change the ordering/sorting of the zones. We assumed it's not important, because the current code doesn't explicitly specific the sort key, so the ordering is probably indeterministic anyway. If that matters, a `.sort(...)` can be added here to pin down the ordering.